code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java

```diff
@@ -114,10 +121,23 @@ private boolean isNetworkUp(long accountId) {
         List<Service> services = objectManager.find(Service.class, SERVICE.ACCOUNT_ID, accountId, SERVICE.REMOVED, null, SERVICE.STACK_ID,
                 networkDriverService.getStackId());
         for (Service service : services) {
-            if (!service.getState().equals(CommonStatesConstants.ACTIVE)) {
+            if (!(service.getState().equals(CommonStatesConstants.ACTIVE) && service.getHealthState().equals(HealthcheckConstants.HEALTH_STATE_HEALTHY))) {
                 return false;
             }
         }
         return true;
     }
+
+    private boolean isNetworkStack(long accountId, long instanceId) {
+        Instance instance = objectManager.findAny(Instance.class, INSTANCE.ID, instanceId);
+        List<? extends Service> services = instanceDao.findServicesFor(instance);
```

Comment:
see if we can change this method in instanceDao to accept instanceid (I bet instanceid is what get used in db query anyway). This way we don't have to load instance on the line 132.